### ISSUE TITLE
Bazel installer should not overwrite ~/.bazelrc

### ISSUE THREAD

**Issue by philwo** [bazelbuild/bazel#2958]
The Bazel installer currently moves ~/.bazelrc to ~/.bazelrc.bak and creates an empty file instead, which is rather annoying. We should change the installer to not touch the file.

**Comment by ittaiz** [comment]
You mean that having a ~/.bazelrc is no longer mandatory so the installer
shouldn't try to create one?

**Comment by philwo** [comment]
@ittaiz I don't think a ~/.bazelrc was ever mandatory, Bazel should work just fine without any configuration files like /etc/bazelrc or ~/.bazelrc. :)

The installer currently always wants to create an empty ~/.bazelrc file. If the user has already created one (and possibly put some config flags there), it will be moved to ~/.bazelrc.bak, which is annoying, because you basically have to immediately move it back.

Instead it should just not touch the file at all - if it doesn't exist already, don't create one. If one exists, leave it as is.

**Comment by softprops** [comment]
On May 6, 2017 4:37 PM, "Philipp Wollermann" <notifications@github.com>
wrote:

@ittaiz <https://github.com/ittaiz> I don't think a ~/.bazelrc was ever
mandatory, Bazel should work just fine without any configuration files like
/etc/bazelrc or ~/.bazelrc. :)

The installer currently always wants to create an empty ~/.bazelrc file. If
the user has already created one (and possibly put some config flags
there), it will be moved to ~/.bazelrc.bak, which is annoying, because you
basically have to immediately move it back.

Instead it should just not touch the file at all - if it doesn't exist
already, don't create one. If one exists, leave it as is.

+1


—
You are receiving this because you are subscribed to this thread.
Reply to this email directly, view it on GitHub
<https://github.com/bazelbuild/bazel/issues/2958#issuecomment-299664397>,
or mute the thread

**Comment by ittaiz** [comment]
Ok. I just don't understand what you meant by "that mechanism is no longer
needed" but I guess it's not that important.

**Comment by philwo** [comment]
@ittaiz No problem and sorry for the misunderstanding. With "that mechanism", I just meant "the part of the installer that deals with the ~/.bazelrc", but re-reading my issue description I can see now how that might be confusing. I removed the part of the sentence. :)
